### ISSUE TITLE
ENV variable loading is fragile

### ISSUE THREAD

**Issue by wildintellect** [MAAP-Project/stac_ipyleaflet#116]
* stac ipyleaflet version: 0.3.3
* Python version: Any
* Operating System: Any

### Description

If you miss any of the required ENV variables in the conda config of the environment or don't specify them early enough in a notebook before trying to load a map then you get errors that are not the easiest to read...
![image (7)](https://github.com/MAAP-Project/stac_ipyleaflet/assets/306179/57045b14-b85b-431e-adb7-03ea29efa7f0)

Ideally we would check/trap these constant loads and issue a clear warning with possible workaround to the users before the Map throws an error. I'm not actually sure why MAAP and VEDA urls are handled differently anyways, can't we just expect a full url supplied every time?

Example:
https://github.com/MAAP-Project/stac_ipyleaflet/blob/2aeda1ba7d938ea94750595e6d0853dfecf4276a/stac_ipyleaflet/stac_discovery/stac_widget.py#L199-L206

**Comment by sandrahoang686** [comment]
We will need to update the constants file so that `STAC_BROWSER_URL` defaults to an empty string as it should be an optional env var. We wont need to check for it then in the checking logic because it will be defaulted.

**Comment by wildintellect** [comment]
@sandrahoang686 but we would need to handle on the UI side what to do if it's blank.

**Comment by chuckwondo** [comment]
I would suggest not using env vars to begin with, in this case. The approach of populating "constants" from env vars is not only very fragile, it also means that it is hard/clumsy to test, and it also means that you cannot have different widgets configured differently.

When using env vars, you want to push their usage to the system interface boundaries. In this case, however, you could probably simply eliminate them altogether.

Of course, I'm not particularly familiar with this code, but I'd suggest eliminating the env vars and adding keyword parameters to the constructor, some of which might have reasonable defaults (e.g., `timeout` and `rescale` could have defaults).

This would allow multiple widgets to be configured differently, if that would be something someone might want to do.

**Comment by wildintellect** [comment]
@chuckwondo I agree that they should be arguments on the constructor, however the defaults vary by platform, e.g. VEDA and MAAP have different defaults and we don't expect users to know all that parameters. So we need some way to set the defaults by platform managers (at install time), with the ability for users to easily override.
